docs/getting_started/import_format.md

```diff
@@ -63,234 +83,173 @@ Using the project template generates already a `manifest.json` file in the proje
 
-      var rowItems = line.Split(": ");
-      var attribute = rowItems[0].Trim();
-      var value = rowItems[1].Trim();
-
-      switch (attribute)
-      {
-        case "Date":
-          measurement.SetAttribute(4,value);
-          break;
-        case "Operator":
-          measurement.SetAttribute(9,value);
-          break;
-      }
-
-    }
-
-    // Parse measured value for each characteristic.
-    while ((line = reader.ReadLine()) != null)
-    {
-      if( string.IsNullOrEmpty(line))
-        continue;
-        
-      var rowItems = line.Split(',');
-      var characteristicName = rowItems[0].Trim();
-      var value = rowItems[1].Trim();
-        
-      var characteristic = root.AddCharacteristic(characteristicName);
-      var measuredValue = measurement.AddMeasuredValue(characteristic);
-      measuredValue.SetAttribute(1,double.Parse(value));
+      case "Text":
+        measurement.SetAttribute(9, value);
+        break;
     }
+  }
 
-    return new ImportData(root);
+  // Parse measured value for each characteristic.
+  while ((line = await reader.ReadLineAsync(cancellationToken)) != null)
+  {
+    var rowItems = line.Split(',', 2);
+
+    if (rowItems.Length < 1)
+      continue;
+    var characteristicName = rowItems[0].Trim();
+    if (string.IsNullOrWhiteSpace(characteristicName))
+      continue;
+    var characteristic = root.AddCharacteristic(characteristicName);
+
+    if (rowItems.Length < 2)
+      continue;
+    var value = rowItems[1].Trim();
+    if (!double.TryParse(value, CultureInfo.InvariantCulture, out var doubleValue))
+      continue;
+                    
+    var measuredValue = measurement.AddMeasuredValue(characteristic);
+    measuredValue.SetAttribute(1, doubleValue);
   }
+  
+  return new ImportData(root);
 }
 ```
 
-The implementation of the example plug-in is now complete. Further information on implementing an import format plug-in can be found in the [Import format]({% link docs/plugin_fundamentals/import_format.md %}).
+We begin with creating a root part and a measurement on this part. This part is the root of all the data we will extract from the import file. During upload, this root part will be merged into the target part by *PiWeb Auto Importer*. The original name of the import target part will be kept during this operation, so the name of the root part does not matter. However, it is good practice to specify a sensible name and we simply use the import file name (without extension) in this example.
+
+Now that we have a part and a corresponding measurement, we can read the import file line by line and add attributes, characteristics and measured values as specified by the import file. Finally we return the root part wrapped in an `ImportData` instance.
+
+<!-- The implementation of the example plug-in is now complete. Further information on implementing an import format plug-in can be found in the [Import format]({% link docs/plugin_fundamentals/import_format.md %}). -->
```

Comment:
What is the comment about, can it be removed or is it still a todo?

Reply:
Removed.